homeassistant/components/dlna_dms/dms.py

```diff
@@ -580,7 +580,8 @@ async def async_browse_search(self, query: str) -> BrowseMediaSource:
             children=children,
         )
 
-        media_source.calculate_children_class()
+        if media_source.children:
```

Comment:
I don't think this check is necessary since `media_source.calculate_children_class` returns early if it has no children.

Reply:
I've removed that.

I also don't understand why it is being called here as it's only used for frontend rendering, doesn't impact the media player.